pyvista/plotting/plotting.py

```diff
@@ -3514,6 +3534,13 @@ def add_mesh(
             else:
                 prop.render_points_as_spheres = render_points_as_spheres
 
+        if scale is not None and style == 'points_gaussian':
+            if isinstance(scale, str):
+                self.mapper.scale_array = scale
+                self.mapper.emissive = emissive
+                self.mapper.scale_factor = 1.0
+
+
```

Comment:
I'm weary of this block. Specifically, I'm weary against having the `scale` argument specifically for the point gaussian mapper and silently ignored for all other scenarios.

Further, if `scale` is a number, should we set `scale_factor` with that value?

Additionally, I'm pretty sure it is unnecessary to set the `emissive` here as it should be set in the `PointGaussianMapper` instantiation on line ~3217 (~3198 head)

Reply:
Thank you for your comments!
- Indeed `scale` is only used with the point gaussian mapper here so I'm creating a new condition here because in this case I need to execute the code from the condition with `render_points_as_spheres` to run `use_circular_splat`. And I didn't want to move too much code.
- For `scale_factor`, do you mean that `scale` could also be a number and work the way you describe?
- About `emissive`, I forgot to explain this, it is set to `True` in the definition of `use_circular_splat`, maybe it should be done differently here?

Reply:
I'm also +1 on the functionality here being added to PyVista, I think the discussion is around what user interface to this should look like.

I am already confused by the interaction and overlap of all the options in `add_mesh`, e.g. when does `point_size` matter?  How will this be different than `scale` in this PR?  Do these interact?  There are `render_points_as_spheres=True/False` and `style=point_guassian`.  I believe the former does something different when the style changes.  IMO the problem will continue to grow unless we address this as new options are added.

Reply:
I think what we're seeing is "feature creep" in the form of adding parameter after parameter to `add_mesh`.

I think the original implementation is flawed, and @kmarchais is merely following our design patten of adding another kwarg to `add_mesh`.

How about this: `add_points_gaussian`. That's explicit and permits us to trim down on args in the already bloaded `add_mesh`.

Reply:
I agree with @akaszynski that we need to rethink the design pattern.  One path forward for this PR is to align on a name for this parameter, and then break it out into a separate method in the future, unless @kmarchais wants to take this part on.  In this case, it would be okay if `scale` is ignored in all other cases.

Reply:
Let's add add this for now in this PR and as a follow up implement `add_points_gaussian`. The only caveat will be that we shouldn't include this in the upcoming release this week, as awesome as this example is.

Reply:
Yes, it still requires some more improvement such as dealing with different types in a similar way as it's done with the `scale` parameter for `glyphs`. Also applying `emissive` here is probably not the best way, but probably not in `use_circular_splat` neither.

Reply:
I have vague memories of someone planning to _reduce_ the number of `add_*()` methods. It may have been @banesullivan? I'm not at all sure I remember correctly, and I can't find any related discussion, so just pointing this out because in case I remember correctly, it's better to short-circuit contradictory fundamental design plans like this.

Reply:
That was me, @adeak -- indeed, I want to reduce the number of `add_*()` methods on the plotter, but mostly in regards to special handling of dataset/mesh types whereas this use case is for special handling of rendering techniques, so I think an `add_point_gaussian()`-like method would be acceptable. Method like `add_lines()` and `add_points()` need to go -- these should be handled by `add_mesh()`. I have a more thorough explanation of my rationale in some comments somewhere.

+1 for creating a clean compromise to merge this PR for release, then rethinking the API design later.

The confliction between `scale` and `point_size` also concerns me... I'm tempted to call this an "advanced" feature and have users set it on the mapper specifically rather than as another `kwarg` in `add_mesh()`. With usage looking like:

```py
actor = plotter.add_mesh(mesh, style='points_gaussian', ...)
actor.mapper.scale_array = 'name of array'
```

Reply:
@banesullivan, I've implemented your suggestion.

All, please verify that this API works. I think it balances everyone's requests.